### PR TITLE
utils: mark as deposit action

### PR DESCRIPTION
**Pull request by jirikuncar**

### CODE REVIEW

**Comment by jirikuncar** [comment]
Alternatively, we can do something like:

```python
def register_action(app, endpoint, action):
    app.add_url_rule(
        '{0}/actions/<any({1}):action>'.format(
            app.config['DEPOSIT_REST_ENDPOINTS'][endpoint]['item_route'],
            action
        ), 
        DepositActionResource.view_name.format(endpoint), 
        app.view_functions['invenio_deposit_rest.{0}_actions'.format(endpoint)]
    )
```

**Comment by egabancho** [comment]
If I am not mistaken this same solution was already proposed, and rejected, some time ago and we all agreed to use `invenio-webhooks` instead of extending the list of actions deposit can do.

cc @CERNDocumentServer

File: invenio_deposit/utils.py [hunk 1]
```diff
@@ -32,6 +32,38 @@
...
+    """Decorator for marking method as deposit action.
+
+    Allows creation of new REST API action on ``Deposit`` subclass.
+    Following example shows possible `cloning` of a deposit instance.
+
+    .. code-block:: python
+
+        from invenio_deposit.api import Deposit
+
+        def CustomDeposit(Deposit):
```
Comment (egabancho) [inline invenio_deposit/utils.py]:
```python
class CustomDeposit(Deposit):
```

**Comment by jirikuncar** [comment]
The alternative solution presented in https://github.com/inveniosoftware/invenio-deposit/pull/160#issuecomment-278376561 is working for CERN Analysis Preservation (kudos @ioannistsanaktsidis). So we don't necessarily need this to be integrated. The bonus point of the decorator solution could be configurable action names, since CAP is using different names for actions in UI.

Another idea is to submit similar PR to Invenio-Records-REST and move the action logic there since, we would like to allow not only cloning of "deposits" but also "records".

**Pull request merged**